.github/actions/install-linux-dependencies/action.yaml

```
@@ -0,0 +1,35 @@
+# Copyright © SixtyFPS GmbH <info@slint-ui.com>
+# SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+# cSpell: ignore libxcb libxkbcommon xfixes
+---
+name: Install Linux dependencies
+description: Set up Linux dependencies for Slint
+
+inputs:
+  extra-packages:
+    description: 'Extra packages to install'
+    required: false
+    default: ""
+  include-gcc:
+    description: 'Include gcc packages (default: "no")'
+    required: false
+    default: "no"
```

I don't recall exactly why we had to force the use of gcc 10 (I'll dig), but I'm pretty sure what's important here is not that we install gcc (that's nothing special) but that we force the use of a specific version. I suggest to reflect that in the input name.

```suggestion
  force-gcc-10:
    description: 'Force use of GCC version 10 (default: "no")'
    required: false
    default: "no"
```

---

Ahh, I suspect because back then gcc 10 wasn't the default but we needed it for C++ support.

---

I think we should keep this for now (but I suggest to rename the input as suggested), and we can remove this altogether when upgrading to a newer Ubuntu version (out of scope for this patch).

---

I did not want to put the version number in: I do expect us to continue setting specific gcc versions going forward and not being happy with the default in Ubuntu :-)

What I wanted to archive here is to make sure we have _one_ place where we consistently set the C/C++ compiler being used. Putting the version into the option name defeats that: You will need to adapt all callers again.

---

> I did not want to put the version number in: I do expect us to continue setting specific gcc versions going forward and not being happy with the default in Ubuntu :-)

I do certainly not expect us to continue with this in the future - it's exactly the opposite. It's in our interest to test and work with the default out-of-the-box system compiler.

Let me rephrase this is with absolute clarify: this choice of gcc 10 is an exception here! That's what it was intended as and I think the name should reflect it.

If you don't want to rename the option, then I'll work separately on removing it altogether again. But please rename it first.

Unless, that is, you can think of a reason for us to devaate from the Ubuntu gcc defaults in the future?

---

OK, I'll rename it.

---

Newest version has this option renamed.

---

Merci :)